src/toolchain.rs

```diff
@@ -605,3 +613,16 @@ impl<'a> Toolchain<'a> {
             .collect())
     }
 }
+
+/// Set the `RUSTUP_TOOLCHAIN_SOURCE` environment variable if not already set.
+///
+/// `RUSTUP_TOOLCHAIN_SOURCE` indicates how the toolchain was determined. The environment
+/// variable could have been set in proxy_mode.rs, in which case it should not be changed.
```

Comment:
Is there a reason  we let the original through rather than override it?

Reply:
I think that's the behavior we want.

In my installation, all the proxies link to rustup:
```
lrwxrwxrwx 1 smoelius smoelius       32 Jan 31  2025 cargo -> /home/smoelius/.cargo/bin/rustup
```
So I think we want it to handle the `+` args and we should just let those through: https://github.com/rust-lang/rustup/blob/061cf7830a61eaefaddddb8454232186d26fd2d1/src/cli/proxy_mode.rs#L19-L26

I am happy to be corrected if I am misunderstanding.

Reply:
`RUSTUP_TOOLCHAIN_SOURCE` should be set to `cli`. But I don't think I understand the point of your question.

Reply:
> If I run with just cargo test instead of cargo +beta test, I observe what you describe.

So my exact example didn't work but the principle still applies.